### DISCUSSION TITLE
Most performative way to keep element fixed in one viewport place while paper is scrolled

### DISCUSSION THREAD

**Discussion by ZalgirisKaunas** [clientIO/joint#2653] [Q&A]
### Introduction

      i need to make some elements fixed in viewport vertically while paper is scrolled. 
i can get the difference of y value and translate the difference but I'm not sure if this is the best way as it's quite slow.
      

### Steps to reproduce

      let lastY = 0;

      this.paperScroller.on('scroll', function(evt) {
        const { x, y } = vue.paperScroller.getVisibleArea()
        const delta =  y - lastY;
        lastY = y;

        vue.graph.getElements().forEach((element) => {
          if (element.attributes.purpose === 'xAxis') {
            element.translate(0, delta);
          }
        });
      });

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
What is the purpose of these elements?

The most efficient way would be to render them above the paper scroller. For example, using HTML and set absolute position and a higher z-index. That way you don't need to update them at all.
